### PR TITLE
General improvements for WEBrick via jekyll serve

### LINKED ISSUE #3791: Jekyll local development over https
Is there a way to run jekyll locally and have it run securely? I'm doing some AJAX posts to a remote server that is https only and I'm getting insecure content errors. Trying to figure out a better way of doing this.

### PR DESCRIPTION
**Pull request by jekyllbot**
PR automatically created for @envygeeks.

Fix #3791/#3478

* Add support for SSL through command line switches.
* Add suppport for file/index.html > file.html > directory.
* Add support for custom-headers through configuration.
* Modernize and split up the serve.
* Add a few basic tests.

### CODE REVIEW

**Comment by envygeeks** [comment]
Woah there captain bot, I think you be needing to just use the short description :P

**Comment by parkr** [comment]
@envygeeks I can change it so it uses just the first line of the commit message as long as it's not "Fix #XXX/#XXX" because that's not useful as a PR title :)

File: lib/jekyll/commands/serve.rb [hunk 1]
```diff
@@ -1,161 +1,198 @@
...
+          "detach" => ["-B", "--detach", "Run the server in the background (detach)"],
+          "port" => ["-P", "--port [PORT]", "Port to listen on"],
+          "host" => ["host", "-H", "--host [HOST]", "Host to bind to"],
+          "skip_initial_build" => ["skip_initial_build", "--skip-initial-build",
+            "Skips the initial site build which occurs before the server " \
+              "is started."],
+
+          "ssl_key" => ["--ssl-key [KEY]", "X.509 (SSL) Private Key."],
+          "ssl_cert" => ["--ssl-cert [CERT]", "X.509 (SSL) certificate."],
+          "baseurl" => ["-b", "--baseurl [URL]", "Base URL"]
```
Comment (parkr) [inline lib/jekyll/commands/serve.rb]: We align at the `=>`

```ruby
"open_url" => ["-o"],
"detach"   => ["-B"],
# ... etc
```

**Comment by envygeeks** [comment]
That's a good idea, because I used that short main message because the more described everything in more detail so I guess it would be better for it to resort to the entire thing in this case.

File: lib/jekyll/commands/serve.rb (hunk 1 above)
Comment (envygeeks) [inline lib/jekyll/commands/serve.rb]: Fixed.

**Comment by parkr** [comment]
Thank you, @envygeeks!

**Pull request merged**